partcad/src/partcad/converters/stl_converter.py

```
@@ -0,0 +1,100 @@
+
+        stl_writer = StlAPI_Writer()
+        stl_writer.ASCIIMode = ascii
+
+        # Ensure the directory exists
+        self.output_file.parent.mkdir(parents=True, exist_ok=True)
+
+        # Try writing the file
+        temp_output = self.output_file.with_suffix(".temp.stl")
+        try:
+            if not stl_writer.Write(simplified_shape, str(temp_output)):
+                raise RuntimeError(f"Failed to write STL file to temporary location: {temp_output}")
```

Ruff (0.8.2)

Abstract `raise` to an inner function

(TRY301)